src/test/java/tools/jackson/databind/jsontype/TestTypedSerializationWithSealedTypes.java

```diff
@@ -0,0 +1,216 @@
+     * class name, written as main-level property name
+     */
+    @Test
+    public void testSimpleClassAsProperty() throws Exception
+    {
+        Map<String,Object> result = writeAndMap(MAPPER, new Cat("Beelzebub", "tabby"));
+        assertEquals(3, result.size());
+        assertEquals("Beelzebub", result.get("name"));
+        assertEquals("tabby", result.get("furColor"));
+        // should we try customized class name?
+        String classProp = Id.CLASS.getDefaultPropertyName();
+        assertEquals(Cat.class.getName(), result.get(classProp));
+    }
+
+    /**
+     * Test inclusion using wrapper style
+     */
+    @Test
+    public void testTypeAsWrapper() throws Exception
+    {
+        ObjectMapper m = jsonMapperBuilder()
+                .addMixIn(Animal.class, TypeWithWrapper.class)
+                .build();
+        Map<String,Object> result = writeAndMap(m, new Cat("Venla", "black"));
+        // should get a wrapper; keyed by minimal class name ("Cat" here)
+        assertEquals(1, result.size());
+        // minimal class name is prefixed by dot, and for inner classes it's bit longer
+        Map<?,?> cat = (Map<?,?>) result.get(".TestTypedSerializationWithSealedTypes$Cat");
+        assertNotNull(cat);
+        assertEquals(2, cat.size());
+        assertEquals("Venla", cat.get("name"));
+        assertEquals("black", cat.get("furColor"));
+    }
+
+    /**
+     * Test inclusion using 2-element array
+     */
+    @Test
+    public void testTypeAsArray() throws Exception
+    {
+        ObjectMapper m = jsonMapperBuilder()
+                .addMixIn(Animal.class, TypeWithArray.class)
+                .build();
+        // hmmh. Not good idea to rely on exact output, order may change. But...
+        Map<String,Object> result = writeAndMap(m, new AnimalWrapper(new Dog("Amadeus", 7)));
+        // First level, wrapper
+        assertEquals(1, result.size());
+        List<?> l = (List<?>) result.get("animal");
+        assertNotNull(l);
+        assertEquals(2, l.size());
+        assertEquals(Dog.class.getName(), l.get(0));
+        Map<?,?> doggie = (Map<?,?>) l.get(1);
+        assertNotNull(doggie);
+        assertEquals(2, doggie.size());
+        assertEquals("Amadeus", doggie.get("name"));
+        assertEquals(Integer.valueOf(7), doggie.get("boneCount"));
+    }
+
+    /* !!! 30-Jan-2010, tatus: I am not completely sure below works as it should
+     *    Problem is, context of "untyped" map should prevent type information
+     *    being added to Animal entries, because Object.class has no type.
+     *    If type information is included, it will not be useful for deserialization,
+     *    since static type does not carry through (unlike in serialization).
+     *
+     *    But it is not quite clear how type information should be pushed through
+     *    array types...
+     */
+    @SuppressWarnings("unchecked")
+    @Test
+    public void testInArray() throws Exception
```

Comment:
As per comment, not 100% sure how this should work so let's just remove test method from here.